hashes/src/sha512_256.rs

```diff
@@ -0,0 +1,205 @@
+// Bitcoin Hashes Library
+// Written in 2018 by
+//   Andrew Poelstra <apoelstra@wpsoftware.net>
+//
+// To the extent possible under law, the author(s) have dedicated all
+// copyright and related and neighboring rights to this software to
+// the public domain worldwide. This software is distributed without
+// any warranty.
+//
+// You should have received a copy of the CC0 Public Domain Dedication
+// along with this software.
+// If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+//
+
+// This module is largely copied from the rust-crypto ripemd.rs file;
+// while rust-crypto is licensed under Apache, that file specifically
+// was written entirely by Andrew Poelstra, who is re-licensing its
+// contents here as CC0.
+
+//! SHA512_256 implementation.
+//!
+
+use core::str;
+use core::ops::Index;
+use core::slice::SliceIndex;
+
+use crate::{hex, sha512, sha512::BLOCK_SIZE, Error};
+
+/// Engine to compute SHA512/256 hash function.
+/// SHA512/256 is a hash function that uses the sha512 alogrithm but it truncates
```

Comment:
There needs to be an "empty" line (containing just `///`) for rustdoc to render the first line as summary. Also now I'm thinking that putting the longer description at module doc would be better. I wouldn't mind putting it in all places (also `Hash` doc - below in the macro).